### PR TITLE
fix: fix analysis csv CLI

### PR DESCRIPTION
**Pull request by bolinocroustibat**
Closes https://github.com/datagouv/hydra/issues/180

- Fix typing issue using minicli for analysis-csv CLI
- Fix method to get record columns in analysis-csv method
- Secure some cases in analysis-csv logic
- Add basic tests for analyse-csv CLI

### CODE REVIEW

File: udata_hydra/cli.py [hunk 1]
```diff
@@ -155,10 +155,11 @@ async def check_resource(resource_id: str, method: str = "get"):
 
 @cli(name="analyse-csv")
 async def analyse_csv_cli(
-    check_id: int | None = None, url: str | None = None, debug_insert: bool = False
+    check_id: str | None = None, url: str | None = None, debug_insert: bool = False
```
Comment (maudetes) [inline udata_hydra/cli.py:158]: Hmm is this a minicli issue? It wouldn't cast it to `int` due to `| None`?

**Review by maudetes — APPROVED** [review]
Thank you for the fix!

File: udata_hydra/cli.py (hunk 1 above)
Comment (bolinocroustibat) [reply udata_hydra/cli.py:158]: No I don't think it was a minicli issue. But all arguments of minicli are strings by default, so it needs to be casted as int (which I did on the fist line of the method). Type hinting argument cannot cast or change the type, it was useless (and false).

Comment (maudetes) [reply udata_hydra/cli.py:158]: I think `size` in `csv_sample` command is used as int directly?

Comment (bolinocroustibat) [reply udata_hydra/cli.py:158]: Good point, my guess is the type hint is wrong, and it's indeed being manipulated in the code as a string.
We should check and fix that.

Comment (bolinocroustibat) [reply udata_hydra/cli.py:158]: OK, just double-checked, you are right, it's seems to be related to the `int | None`, which seems to changes minicli's casting behaviour. It seems to be a bug from minicli. We already had a very similar bug before. I will created a ticket.

Comment (bolinocroustibat) [reply udata_hydra/cli.py:158]: OK thanks for spotting, I just created an issue: https://github.com/pyrates/minicli/issues/15

In the meantime we can continue using hydra CLI as it is.

**Pull request merged**